code/modules/power/apc/apc_main.dm

```diff
@@ -206,10 +206,14 @@
 		disconnect_terminal()
 	. = ..()
 
-/obj/machinery/power/apc/handle_atom_del(atom/atom_to_check)
-	if(atom_to_check == cell)
+/obj/machinery/power/apc/handle_atom_del(atom/deleting_atom)
+	if(deleting_atom == cell)
 		cell = null
+		charging = APC_NOT_CHARGING
 		update_appearance()
+		if(!QDELING(src))
+			SStgui.update_uis(src)
```

Comment:
``handle_atom_del`` is called on ``Destroy``, in what case would this be needed?

Reply:
handle_atom_del is called when something CONTAINED in an atom is destroyed, ie if an admin deletes the cell via the view variables panel. 
![image](https://user-images.githubusercontent.com/6209658/186483620-a8d8d9da-b96b-4428-b9a1-82b047490393.png)